src/app/action/action.controller.js

```diff
@@ -1,21 +1,46 @@
 export class ActionController {
-  constructor($firebaseArray) {
+    constructor($firebaseArray, $stateParams) {
+
     'ngInject';
     this.person='';
     this.items='';
     this.text='';
-    this.roomID='101'; //temporary
-    this.assignee=['mayuresh','abhimanyu','aishwarya'];
-    this.myDataRef = new Firebase('https://vyapi.firebaseio.com/action');
+    this.assignee=[];
+    var self=this.assignee;
+
+    this.roomID=$stateParams.roomKey; //temporary
+    var roomRef = new Firebase('https://vyapi.firebaseio.com/rooms/'+this.roomID);
+    var room = roomRef.child("members");
+    room.once("value", function(snapshot) {
+    var data1 = snapshot.forEach(function(uid){
+      var personRef = new Firebase('https://vyapi.firebaseio.com/users/');
+      //console.log(''+child.key()+'/google/');
+      var data2=personRef.child(''+uid.key()+'/google/');
+      $firebaseArray(data2);
+      data2.once('value',function(userSnapshot){
```

Comment:
use fat arrow function , that would save you from using `var self = this.assigne`. consult @monicagangwar and @diwakar-s-maurya for their implementation